v3/src/hooks/use-keyboard-shortcuts.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from "react"
+import { tinykeys } from "tinykeys"
+import { appState } from "../models/app-state"
+
+export const useKeyboardShortcuts = () => {
+  useEffect(function handleKeyboardShortcuts() {
+
+    const isEditable = (el: Element | null) => {
+      if (!el) return false
+      if (el instanceof HTMLInputElement ||
+        el instanceof HTMLTextAreaElement ||
+        (el instanceof HTMLElement && el.isContentEditable)) {
+        return true
+      }
+    }
+
+    const okToUndoRedo = (e: KeyboardEvent, test:boolean) => {
+      const target = e.target as Element | null
+      if (isEditable(target)) return false
+      e.preventDefault()
```

Comment:
1. You probably want to `stopPropagation()` as well as `preventDefault()`, at least if we're handling the event.
2. I'm wondering whether we should only `preventDefault()` and `stopPropagation()` if `test` is `true`. 🤔  The argument _for_ doing so is that if we're not going to handle the event perhaps we should let the browser (or someone else) have a crack at it. On the other hand, we're not using a capturing handler and we're attached to the window, so we're getting the event fairly late in the process and maybe we don't want anyone behind us to make use of one of our keyboard shortcuts anyway. 🤷‍♂️